### PR TITLE
Vet Clinic : Milestone #4 

### PR DESCRIPTION
**Pull request by ahmed025566**
In this project you can see that I have made the following: 

I have created vets table and inserted vets data into it.
I have created specializations table and inserted vets id and specializations ids data into it.
I have created visits table and inserted some visits data into it.
I have written the attached queries to answer required questions.

T1

T2

T3

T4
rc

T6

T7

T8

T9

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -125,3 +125,110 @@ ON owner_id = owners.id
...
+
+--What animal has the most visits to vets?
+SELECT
+  animals.name AS animal,
+  COUNT(visits.animal_id) AS number_of_visits
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+GROUP BY animals.name
+ORDER BY COUNT(visits.animal_id) DESC
+LIMIT 1;
```
Comment (muneebulrehman) [inline queries.sql:181]: As per the screenshot you have attached for this query, it is returning the wrong results. The result of this query is returning `Squirtle` which is in fact the least visiting animal. Kindly fix this issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @ahmed025566 , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive PR
- Included screenshots
- New tables created successfully.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 2]
```diff
@@ -125,3 +125,110 @@ ON owner_id = owners.id
...
+SELECT vets.name AS vet_name,
+  animals.name AS animal_name,
+  visit_date AS last_visit
+FROM vets
+JOIN visits ON vets.id = vet_id
+JOIN animals ON animal_id = animals.id
+WHERE vets.name = 'William Tatcher'
+GROUP BY vets.name, animals.name, visit_date
+ORDER BY MAX(visit_date) DESC
+LIMIT 1;
```
Comment (muneebulrehman) [inline queries.sql:139]: As per the screenshot you have attached for this query, it is returning the wrong results. The result of this query is returning `Agumon` which is not the last animal seen by William Tatcher. Kindly fix this issue.

File: queries.sql [hunk 3]
```diff
@@ -125,3 +125,110 @@ ON owner_id = owners.id
...
+
+--How many visits were with a vet that did not specialize in that animal's species?
+SELECT
+  COUNT(visits.animal_id) AS visits
+FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.id
+JOIN specializations ON vets.id = specializations.vet_id
+JOIN species ON specialization_id = species.id
+WHERE specialization_id != animals.species_id;
```
Comment (muneebulrehman) [inline queries.sql:222]: As per the screenshot attached for this query, it is returning the wrong results. The actual result should be 12 while it is returning 7. Kindly fix this issue.

File: queries.sql (hunk 2 above)
Comment (Davidosky007) [reply queries.sql:139]: * As mentioned by the previous CR this issue hasn't been fixed Kindly run your queries again i attach the correct screenshots

File: queries.sql (hunk 3 above)
Comment (Davidosky007) [reply queries.sql:222]: * As mentioned by the previous CR this issue hasn't been fixed Kindly run your queries again i attach the correct screenshots

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ahmed025566 👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Tables created properly 👍 
* Well written PR 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**